crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -452,13 +437,23 @@ pub trait WorldQueryGats<'world> {
 pub unsafe trait ReadOnlyWorldQuery: WorldQuery<ReadOnly = Self> {}
 
 /// The `Fetch` of a [`WorldQuery`], which is used to store state for each archetype/table.
-pub type QueryFetch<'w, Q> = <Q as WorldQueryGats<'w>>::Fetch;
+#[deprecated(note = "Use `Q::Fetch<'w>` instead", since = "0.9")]
+pub type QueryFetch<'w, Q> = <Q as WorldQuery>::Fetch<'w>;
```

Comment:
I don't think this is a particularly prominent user-facing, probably better to just remove this and add it to the migration notes.

Same with the other aliases.